### PR TITLE
PLT-6050 Prompt user when leaving System Console page with unsaved changes

### PR DESCRIPTION
**Pull request by hmhealey**
This is a reresubmission of https://github.com/mattermost/platform/pull/7018 rebased onto master and with my review feedback addressed. Original PR description below:

#### Summary
The user is prompted with a "discard changes" dialog when they have unsaved changes and attempt the following:

- navigating to another console page
- switching teams
- logging out

The dialog gives them the option to cancel their navigation, or confirm it.

Implementation details:
- Created a more general system for blocking navigation pending confirmation. This approach could eventually be expanded to confirm navigation (from unsaved changes or other things) in other parts of the application.
- Instead of using `connect` a lot or passing actions through multiple components, created a new connected container component for links, `BlockableLink` to encapsulate the navigation confirmation logic. Since `AdminNavbarDropdown` didn't use a `Link`, I connected it.
- Introduced redux-thunk action creator
- The mockup had a dialog with blue buttons, whereas the theme for the System Console resulted in a grey `btn-default`. I tracked down the CSS that made other buttons blue. It was in utils.jsx for theming, and I decided it was too difficult to work with without having unintended consequences. I made my own CSS for the discard changes modal.
- Changed the linting rules so I could use spread rest syntax

#### Ticket Link
https://mattermost.atlassian.net/browse/PLT-6050
https://github.com/mattermost/platform/issues/6230

#### Checklist
- [x] Has UI changes
- [x] Includes text changes and localization file ([.../i18n/en.json]

### CODE REVIEW

File: components/admin_console/blockable_link/blockable_link.jsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React from 'react';
+import ReactDOM from 'react-dom';
+import PropTypes from 'prop-types';
+
+import {Link, browserHistory} from 'react-router/es6';
+
+export default class BlockableLink extends React.Component {
```
Comment (jwilander) [inline components/admin_console/blockable_link/blockable_link.jsx]: Should this be pure?

File: components/admin_console/blockable_link/blockable_link.jsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            /*
+             * Function for deferring navigation while blocked
+             */
+            deferNavigation: PropTypes.func.isRequired
+        }).isRequired
+    }
+
+    constructor(props) {
+        super(props);
+        this.handleClick = this.handleClick.bind(this);
```
Comment (jwilander) [inline components/admin_console/blockable_link/blockable_link.jsx]: Why not use arrow function syntax for binding?

**Comment by hmhealey** [comment]
I'll do a quick style pass and make sure we're using PureComponents wherever possible

**Comment by hmhealey** [comment]
@crspeller Can we get a review on this before I rebase it again?

**Pull request merged**